### PR TITLE
Round modifiers: a new admin panel for changing aspects of the round

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
admin: Added the Round Modifiers panel, a more streamlined way for admins to change aspects of the round.
/:cl:

This is very much a tool that I wish existed for me as an admin. You will often be running events, and mass var edit everyone/thing on the station, but then you'll have to reapply when new people late-join. So, this is a way of ensuring that you can apply STUFF to people both on roundstart, and latejoins.

image

Let's quickly talk through the three example "round modifiers" that I've coded up.

- Mass Pacifism, applies the PACIFIST trait to all mobs constantly, meaning a universe mostly (because pacifism isn't perfect) free of physical violence.
- Silicon Law Immutability, right now, disables the ability to change AI laws through AI Upload consoles. Note that EVERY OTHER method of changing isn't disabled. I envisioned this specifically for when I was running a nations round, and needed people to stop uploading laws to the neutral AI.
- Anime Station, a reimplementation of the Secrets->Chinese Cartoons button, BUT this can be triggered before the round starts, will apply to everyone as the game starts (even dem nuke ops), and late joins. Also, it renames silicons. And Beepsky-kun

Round modifiers should be be generally able to be toggled, although I acknowledge that some will be one-way.

In summary, this PR is about the system, and here's some light content to demonstrate it.

### CODE REVIEW

**Comment by subject217** [comment]
yuh yikies another menu with the laggy tgui buttons

**Comment by coiax** [comment]
The difference is datums. Difference between anime button on secrets panel and anime button on round modifier panel is that it'll hang around, and anime-up late joiners.

**Comment by optimumtact** [comment]
Long term, most of what is in secrets/fun and trigger events should move to this panel as well fyi.

**Comment by OverlyGeneric** [comment]
> Long term, most of what is in secrets/fun and trigger events should move to this panel as well fyi.

I see no reason why OP shouldn't start moving the secrets to the panel before this gets merged.

**Comment by JJRcop** [comment]
I think you should consider supporting categories, like the uplink has.

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -363,6 +364,16 @@ SUBSYSTEM_DEF(ticker)
 				to_chat(N, "Captainship not forced on anyone.")
 			CHECK_TICK
 
+/datum/controller/subsystem/ticker/proc/apply_round_modifiers()
+	for(var/mob/dead/new_player/N in GLOB.player_list)
+		var/mob/living/L = N.new_character
+		if(!istype(L))
+			continue
+
+		for(var/d in SSround_modifiers.active_modifiers)
```
Comment (Jordie0608) [inline code/controllers/subsystem/ticker.dm]: Check if there are any active modifiers first, stops from looping through all players only to do nothing.

File: code/datums/round_modifiers/_round_modifier.dm [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+/datum/round_modifier/proc/on_apply()
+	return
+
+/datum/round_modifier/proc/on_remove()
+	return
+
+/datum/round_modifier/proc/announce_enabling()
+	priority_announce("A bug has occured with the universe. Please notify a Centcom employee immediately.")
+
+/datum/round_modifier/proc/announce_disabling()
```
Comment (Jordie0608) [inline code/datums/round_modifiers/_round_modifier.dm]: Unless I'm blind these two procs aren't called anywhere, intentional?

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

File: code/datums/round_modifiers/_round_modifier.dm (hunk 2 above)
Comment (coiax) [reply code/datums/round_modifiers/_round_modifier.dm]: Yeah, they're not. In a hypothetical future, when enabling/disabling, Centcom messages could be sent to tell people IC about the various rules. But that's not implemented right now.

Comment (Jordie0608) [reply code/datums/round_modifiers/_round_modifier.dm]: You've got two modifiers with announcements defined, implement it =P

**Comment by ExcessiveUseOfCobblestone** [comment]
@coiax

**Comment by coiax** [comment]
I don't have free time to code anything even remotely complex anymore.
